src/Microsoft.AspNet.Razor/RazorError.cs

```diff
@@ -46,19 +61,23 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var err = obj as RazorError;
-            return (err != null) && Equals(err);
+            var error = obj as RazorError;
+            return Equals(error);
         }
 
         public override int GetHashCode()
         {
-            return base.GetHashCode();
+            return HashCodeCombiner.Start()
+                .Add(Message, StringComparer.Ordinal)
+                .Add(Location)
```

Comment:
We should be comparing it. An error with unequal `Length`'s is not the same error.